File: src/NewsPortal.Client/src/types/NewsSource.ts
export interface NewsSource {
    id: number;
    name: string;
    slug: string;
    baseUrl: string;
    logoUrl?: string;
    fetchMethod: number; // 1: Rss, 2: Api, 3: Scrape
    rssFeedUrl?: string;
    apiEndpoint?: string;
    apiKey?: string;
    fetchIntervalMinutes: number;
    isActive: boolean;
    lastFetchedAt?: string;
    healthStatus?: number; // 0: Active, 1: Degraded, 2: Paused, 3: Disabled
    consecutiveFailures?: number;
    lastSuccessAt?: string;
    lastFailureAt?: string;
    lastErrorCode?: string;
    nextRetryAt?: string;
    requestTimeoutSeconds?: number;
    maxRetryAttempts?: number;
    circuitBreakerThreshold?: number;
    articleCount?: number;
}

export interface CreateNewsSourceDto {
    name: string;
    baseUrl: string;
    logoUrl?: string;
    fetchMethod: number;
    rssFeedUrl?: string;
    apiEndpoint?: string;
    apiKey?: string;
    fetchIntervalMinutes: number;
}

export interface NewsSourceTestIssue {
    code: string;
    message: string;
    method?: string;
}

export interface NewsSourceTestResult {
    isSuccess: boolean;
    message: string;
    primaryMethod: number;
    successfulMethod?: number;
    usedFallback: boolean;
    articlesFetched: number;
    validArticles: number;
    invalidArticles: number;
    sampleTitles: string[];
    issues: NewsSourceTestIssue[];
    durationMs: number;
}

export interface BulkNewsSourceActionRequest {
    sourceIds: number[];
    action: 'pause' | 'resume' | 'fetch';
}

export interface BulkNewsSourceActionResult {
    action: string;
    totalRequested: number;
    affectedCount: number;
    queuedJobs: number;
    skippedSourceIds: number[];
    message: string;
}

export interface FetchJobResponse {
    message: string;
    sourceId: number;
    jobId: string;
    status: string;
    hangfireJobId?: string;
}

export interface FetchJobStatusResponse {
    jobId: string;
    sourceId: number;
    sourceName: string;
    status: string;
    triggerType: string;
    attempts: number;
    startedAt?: string;
    finishedAt?: string;
    articlesFetched: number;
    newArticles: number;
    updatedArticles: number;
    errorCode?: string;
    errorSummary?: string;
}

File: src/NewsPortal.Core/Entities/NewsSource.cs
using NewsPortal.Core.Enums;

namespace NewsPortal.Core.Entities;

public class NewsSource : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string BaseUrl { get; set; } = string.Empty;
    public string? LogoUrl { get; set; }
    public FetchMethod FetchMethod { get; set; } = FetchMethod.Rss;
    public string? RssFeedUrl { get; set; }
    public string? ApiEndpoint { get; set; }
    public string? ApiKey { get; set; }
    public int FetchIntervalMinutes { get; set; } = 30;
    public DateTime? LastFetchedAt { get; set; }
    public SourceHealthStatus HealthStatus { get; set; } = SourceHealthStatus.Active;
    public int ConsecutiveFailures { get; set; }
    public DateTime? LastSuccessAt { get; set; }
    public DateTime? LastFailureAt { get; set; }
    public string? LastErrorCode { get; set; }
    public string? LastErrorMessage { get; set; }
    public DateTime? NextRetryAt { get; set; }
    public int RequestTimeoutSeconds { get; set; } = 90;
    public int MaxRetryAttempts { get; set; } = 3;
    public int CircuitBreakerThreshold { get; set; } = 5;

    // Navigation
    public virtual ScrapingConfig? ScrapingConfig { get; set; }
    public virtual ICollection<NewsArticle> Articles { get; set; } = new List<NewsArticle>();
    public virtual ICollection<SourceFetchJob> FetchJobs { get; set; } = new List<SourceFetchJob>();
}
